### ISSUE TITLE
Support MySQL GRANT / CREATE USER statements

### ISSUE THREAD

**Issue by acdha** [sqlfluff/sqlfluff#3188]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

We have some SQL files which define service users and their privileges. Currently sqlfluff will report unparsable sections for each of these lines as well as common variations (e.g. using `IDENTIFIED BY 'password'` instead of the AWS RDS IAM extension, granting only a single permission, etc.:

```sql
CREATE USER 'prj_svc' IDENTIFIED WITH AWSAuthenticationPlugin AS 'RDS';
GRANT INSERT, UPDATE, DELETE, SELECT, REFERENCES ON prj_table TO 'prj_svc';
FLUSH PRIVILEGES;
```

### Use case

We're following the process here: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.DBAccounts.html#UsingWithRDS.IAMDBAuth.DBAccounts.MySQL

In a larger context, this is similar to #901 where I'd like to be able to have sqlfluff enabled for *.sql without exclude rules.

### Dialect

MySQL

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
